src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,308 @@ const ActorInit Bg_Mizu_Water_InitVars = {
+                this->targetY = this->baseY + 85.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[6].unk_02 = this->actor.posRot.pos.y;
+            break;
+        case 3:
+            if (Flags_GetSwitch(globalCtx, this->switchTriggerID) != 0) {
+                this->targetY = this->baseY + 110.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[8].unk_02 = this->actor.posRot.pos.y;
+            break;
+        case 4:
+            if (Flags_GetSwitch(globalCtx, this->switchTriggerID) != 0) {
+                this->targetY = this->baseY + 160.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[16].unk_02 = this->actor.posRot.pos.y;
+            break;
+    }
+
+    if (this->targetY < this->actor.posRot.pos.y) {
+        func_800AA000(0.0f, 0x78, 0x14, 0xA);
+        func_8002F948(&this->actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    } else if (this->targetY > this->actor.posRot.pos.y) {
+        func_800AA000(0.0f, 0x78, 0x14, 0xA);
+        func_8002F948(&this->actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    }
+}
+
+void BgMizuWater_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuWater* this = THIS;
+    s32 y;
+    s32 unk0;
+    s32 unk1;
+    u32 pad;
+
+    if (bREG(15) == 0) {
+        osSyncPrintf("%x %x %x\n", Flags_GetSwitch(globalCtx, 0x1C), Flags_GetSwitch(globalCtx, 0x1D),
+                     Flags_GetSwitch(globalCtx, 0x1E));
+    }
+
+    if (this->type == 0) {
+        y = this->actor.posRot.pos.y;
+        unk0 = 0;
+        unk1 = 0;
+        if (y < -15.0f) {
+            unk0 = 0;
+            unk1 = ((y - -835.0f) / 820.0f) * 200.0f;
+        } else if (y < 445.0f) {
+            unk0 = 1;
+            unk1 = 0xFF - (s32)(((y - -15.0f) / 460.0f) * 95.0f);
+        } else if (y <= 765.0f) {
+            unk0 = 2;
+            unk1 = 0xFF - (s32)(((y - 445.0f) / 320.0f) * 95.0f);
+        }
+        globalCtx->unk_11D30[1] = ((u8)unk0 << 8) | (unk1 & 0xFF);
+    }
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgMizuWater_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuWater* this = THIS;
+    s32 gameplayFrames;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
```

Comment:
We just merged a change to how we handle graphics functions, so you'll want to adapt your GFX funcs to the new style. See other actors for reference :)